src/Form/matching.js

```diff
@@ -805,10 +814,19 @@ const getRelatedText = (el, form, cssSelector) => {
     // If there is still no meaningful container return empty string
     if (scope === el || scope.nodeName === 'SELECT') return ''
 
-    // If the container has a select element, remove its contents to avoid noise
-    const text = removeExcessWhitespace(extractElementStrings(scope).join(' '))
+    let trimmedText = ''
+    const label = scope.querySelector('label')
+    if (label) {
+        // Try searching for a label first
+        trimmedText = removeExcessWhitespace(label.textContent)
```

Comment:
If there's a label that's not been explicitly tied to the field (no `for="id"`), try reading that first to avoid other noisy text.

Reply:
Worth using `getText` here? (it may need updated to work with label elements)